### ISSUE TITLE
"rules is not defined" - custom config

### ISSUE THREAD

**Issue by tomdavidson** [SparkartGroupInc/quality-docs#3]
Thank you for this project. I am trying to put it to work and am stuck on using a config file.  For example:

``` js
$ quality-docs {,**/}*.md --config quality-docs.json --ignore BYU
/home/tom/.nvm/versions/node/v6.5.0/lib/node_modules/quality-docs/index.js:78
      fs.writeFile(cli.flags.rules, JSON.stringify(rules, null, 2), function(err) {
                                                   ^

ReferenceError: rules is not defined
    at Object.<anonymous> (/home/tom/.nvm/versions/node/v6.5.0/lib/node_modules/quality-docs/index.js:78:52)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Module.runMain (module.js:590:10)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:509:3
```

and quality-docs.json

``` js
{
    "dictionaries": " ",
    "rules": {
        "filler": {
            "severity": "warn"
        },
        "hedge": {
            "severity": "suggest"
        },
        "list-item-content-indent": false,
        "no-auto-link-without-protocol": false,
        "retext-readability": {
            "age": 18,
            "minWords": 7,
            "severity": "fatal"
        },
        "weasel": {
            "severity": "suggest"
        }
    },
    "ignore": [

    ],
    "noIgnore": [

    ]
}
```

**Comment by tomdavidson** [comment]
a config file of :

``` js
{
    "ignore": [
        "BYU"
    ]
}
```

Throws fits too:

```
$ quality-docs {,**/}*.md --config quality-docs.json 
/home/tom/.nvm/versions/node/v6.5.0/lib/node_modules/quality-docs/index.js:96
  customConfig.dictionaries.forEach((dictionaryPath) => {
                           ^

TypeError: Cannot read property 'forEach' of undefined
    at Object.<anonymous> (/home/tom/.nvm/versions/node/v6.5.0/lib/node_modules/quality-docs/index.js:96:28)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Module.runMain (module.js:590:10)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:509:3
```

Perhaps I misunderstand the basic usage are certain elements (such as a dictionary) required if I use a config file?

**Comment by ianwremmel** [comment]
Seeing the same issue. I tried adding a "dictionaries" property and got a different error (probably because I'm not clear what a valid path to the default dictionary is).

**Comment by localjo** [comment]
Sorry for the delayed response. I wasn't getting notifications for this repo for some reason. I'll take a look at this tomorrow.

**Comment by localjo** [comment]
@tomdavidson @ianwremmel Sorry for the long delay on this. This error is fixed in v3.1.2

`npm i -g quality-docs@3.1.2` should get you the update. Just a note, while an empty string for the dictionaries key should be fine, a single space, ie `"dictionaries": " "` will not work, since a space is not a valid dictionary filename. If you're not providing a custom dictionary, I'd recommend removing the key entirely from your config file.

**Comment by tomdavidson** [comment]
Thank you @josiahsprague !
